mlflow/server/js/src/experiment-tracking/components/App.js

```diff
@@ -85,6 +86,7 @@ class App extends Component {
               <Route exact path={Routes.metricPageRoute} component={MetricPage} />
               <Route exact path={Routes.compareRunPageRoute} component={CompareRunPage} />
               <Route path={Routes.experimentPageSearchRoute} component={HomePage} />
+              <Route exact path={Routes.authRoute} component={AuthComponent} />
```

Comment:
Still i dont now how the react router works but if you can implements a guard (for route where you know for sure where you need auth), you will avoid delayed jump to the SSO (as for Perpetuo for example...)

Reply:
Only the flask application need auth, so the only routes that need auth are the one making backend calls. What do you mean by a guard ?

Reply:
A mean a check that auth is present before trying to load the page (route) in which you are sure that you will need auth. to avoid the scenario: The page is loading (and so the UI is changing) and oups you catch a 401 and redirect to the SSO.
To do this kind of gard, you can call the is_auth endpoint

Reply:
I agree this would be a useful feature, but are you ok if we do this in another review ?